src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -353,10 +427,15 @@ private IEnumerable<string> GetAttributeNames(Block tagBlock)
                     }
                 }
 
-                attributeNames.Add(childSpan.Content.Substring(start, end - start));
+                var attributeName = childSpan.Content.Substring(start, end - start);
+                var attributeValue = _attributeValueBuilder.ToString();
+                var attribute = new KeyValuePair<string, string>(attributeName, attributeValue);
+                attributes.Add(attribute);
+
+                _attributeValueBuilder.Clear();
```

Comment:
This ones in the `for` loop though and done X many attribute times where as the other is only done once per method call.